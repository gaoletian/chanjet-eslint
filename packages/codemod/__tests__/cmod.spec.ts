import { execSync } from 'child_process';
import path from 'path';

const execCmd = (cmdMock: string) => {
  const cmd = cmdMock.replace('cmod', `node ${path.join(__dirname, '../dist/cmod.js')}`);
  return execSync(cmd, { encoding: 'utf8' });
};

describe('cmod command test', () => {
  const helperInfo = 'cmod [ tosrc|eslint ] <options>';
  test('should show help', () => {
    expect(execCmd('cmod')).toContain(helperInfo);
    expect(execCmd('cmod ')).toContain(helperInfo);
    expect(execCmd('cmod --help')).toContain(helperInfo);
  });
  test('cmod --dry', () => {
    expect(execCmd('cmod src "packages/codemod/__tests__/fixture/**/*" --dry')).toMatchSnapshot();
    expect(execCmd('cmod src --dry')).toMatchSnapshot();
    expect(execCmd('cmod src --cached --dry')).toMatchSnapshot();
    expect(execCmd('cmod eslint "packages/codemod/__tests__/fixture/**/*" --dry')).toMatchSnapshot();
    expect(execCmd('cmod eslint --dry')).toMatchSnapshot();
    expect(execCmd('cmod eslint --cached --dry')).toMatchSnapshot();
  });
});
