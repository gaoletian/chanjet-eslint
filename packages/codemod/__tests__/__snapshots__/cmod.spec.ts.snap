// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cmod command test cmod --dry 1`] = `"{\\"filePaths\\":[\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/jsfile.js\\",\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/jsxfile.jsx\\",\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/tsfile.ts\\",\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/tsxfile.tsx\\"],\\"currentCmd\\":{\\"title\\":\\"修复模块导入路径\\",\\"path\\":\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/dist/commands/src-alias.js\\"}}"`;

exports[`cmod command test cmod --dry 2`] = `"{\\"filePaths\\":[],\\"currentCmd\\":{\\"title\\":\\"修复模块导入路径\\",\\"path\\":\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/dist/commands/src-alias.js\\"}}"`;

exports[`cmod command test cmod --dry 3`] = `"{\\"filePaths\\":[],\\"currentCmd\\":{\\"title\\":\\"修复模块导入路径\\",\\"path\\":\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/dist/commands/src-alias.js\\"}}"`;

exports[`cmod command test cmod --dry 4`] = `"{\\"filePaths\\":[\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/jsfile.js\\",\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/jsxfile.jsx\\",\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/tsfile.ts\\",\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/__tests__/fixture/tsxfile.tsx\\"],\\"currentCmd\\":{\\"title\\":\\"eslint fix-import-type\\",\\"path\\":\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/dist/commands/eslint.js\\"}}"`;

exports[`cmod command test cmod --dry 5`] = `"{\\"filePaths\\":[],\\"currentCmd\\":{\\"title\\":\\"eslint fix-import-type\\",\\"path\\":\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/dist/commands/eslint.js\\"}}"`;

exports[`cmod command test cmod --dry 6`] = `"{\\"filePaths\\":[],\\"currentCmd\\":{\\"title\\":\\"eslint fix-import-type\\",\\"path\\":\\"/Users/gaoletian/chanjet/chanjet-eslint/packages/codemod/dist/commands/eslint.js\\"}}"`;
